flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/slotpool/DeclarativeSlotPoolService.java

```diff
@@ -134,7 +139,25 @@ public final void start(
      *
      * @param componentMainThreadExecutor componentMainThreadExecutor used by this slot pool service
      */
-    protected void onStart(ComponentMainThreadExecutor componentMainThreadExecutor) {}
+    protected void onStart(ComponentMainThreadExecutor componentMainThreadExecutor) {
```

Comment:
This check logic already exists in its subclass(i.e. `DeclarativeSlotPoolBridge`), we'd better make some changes to the subclass to reduce duplication of code.